### DISCUSSION TITLE
Practical Modal Example

### DISCUSSION THREAD

**Discussion by jmdamato** [react-hook-form/react-hook-form#4336] [Q&A]
I have a long form and would like to use a modal to add up to 4 additional people's info to this form using an "add friend" button that will bring up a modal. I'm looking at the example on the site https://codesandbox.io/s/react-hook-form-conditional-inputs-c7n0r for Modals, but it doesn't make much sense to me. Can this exact functionality be done without showing the original **mail** input? 

I essentially need a modal that has form fields that are an extension to the original form which I will then upload to a database.

**Comment by bluebill1049** [comment]
>  Can this exact functionality be done without showing the original mail input?

what do you mean?

**Reply by jmdamato** [reply]
Hi bluebill, IMO I was thinking a modal should add to the form ex)

![image](https://user-images.githubusercontent.com/5658718/109816511-d297b780-7bfe-11eb-844b-b21383f87049.png)

{toggle: true, mail: "...", keepValue: "...", ghost: "...", **foo: "...", bar: "..."**}

**Reply by bluebill1049** [reply]
what do you mean by "add to the form ex"?

**Reply by jmdamato** [reply]
ex - example. I uploaded an image of what I am trying to do.
The current example has a modal that updates the mail input, it does not append to the list of form inputs. I would like a modal with additional inputs (foo and bar) that get added to the form inputs if given a value. If foo or bar are not given a value, then they would either not exist in the list of form inputs or have a value of "".

**Reply by bluebill1049** [reply]
you want to `append`? sounds like `useFieldArray`? and if you want data not lost when modal get unmounted, use `shouldUnregister: false`

**Reply by jmdamato** [reply]
Thanks for all the help @bluebill1049 ! I have it working mostly now just using shouldUnregister: false, but it's a little clunky. For example values of "" will show up if you don't fill it the modal and close it. Also if you fill in USER 3, then it create an array [undef, undef, Object]. Do you have any pointers on how to optimize this? I tried looking into useFieldArray, but was not sure how to utilize it on the SAVE CHANGES button after the user puts in their info.


https://codesandbox.io/s/react-hook-form-usefieldarray-forked-27br3?file=/src/index.js

**Reply by bluebill1049** [reply]
I think would be cleaner if you divide into two separate forms, one form `handleSubmit` and inject data into your data source

**Reply by jmdamato** [reply]
Do you have an example of what you mean by inject? 

I'm trying to do:
```
  React.useEffect(() => {
    console.log('EFFECTS')
    append(modalFormData)
  }, [append, modalFormData]);
```

But it's not updating the original form (the modal data gets lost) I also have `shouldUnregister: false`

**Reply by bluebill1049** [reply]
maybe this example helps: https://codesandbox.io/s/tabs-760h9 with multiple forms
